tests/test_util.py

```diff
@@ -32,3 +33,37 @@ def test_find_closest_point():
     actual_distances, actual_index = find_closest_point([-99.83, -99.32], [42.25, 42.21], -99.32, 43.25)
     assert_array_almost_equal(actual_distances, expected_distances)
     assert actual_index == expected_index
+
+
+def test_to_absolute_path():
+    input_path = "~/nonexistent_file.txt"
+    parsed = to_absolute_path(input_path)
+    expected = Path.home() / "nonexistent_file.txt"
+    assert parsed == expected
+
+
+def test_to_absolute_path_must_exist():
+    input_path = "~/nonexistent_file.txt"
+    with pytest.raises(ValueError) as excinfo:
+        to_absolute_path(input_path, must_exist=True)
+    assert "Got non-existent path" in str(excinfo.value)
+
+
+def test_to_absolute_path_with_absolute_input_and_parent(tmp_path):
+    input_path = tmp_path / "nonexistent_file.txt"
+    parsed = to_absolute_path(input_path, parent = tmp_path)
```

Comment:
My editor complains about types
```suggestion
    parsed = to_absolute_path(str(input_path), parent = tmp_path)
```